src/compile/Component.ts

```diff
@@ -541,9 +541,21 @@ export default class Component {
 	}
 
 	walk_module_js() {
+		const component = this;
 		const script = this.ast.module;
 		if (!script) return;
 
+		walk(script.content, {
+			enter(node) {
+				if (node.type === 'LabeledStatement' && node.label.name === '$') {
+					component.warn(node, {
+						code: 'module-script-reactive-declaration',
+						message: '$: has no effect in a module script'
+					});
+				}
+			}
+		});
+
```

Comment:
It didn't feel right to check for reactive declarations in either `add_sourcemap_locations` or `remove_indentation`, so after some discussing in chat we decided to make a separate walk for this.